src/docs/automation-and-resiliency/application-resource-tuning.md

```diff
@@ -0,0 +1,290 @@
+
+**Resource Requests**  
+Resource requests are guaranteed and reserved for the pod. _Scheduling decisions are made based on the request_ to ensure that a node has enough capacity available to meet the requested value.
+
+**Resource Limits**  
+Resource limits set an upper limit of what a pod can burst to if the resources are available on the node.
+
+## [](https://developer.gov.bc.ca/Developer-Tools/Resource-Tuning-Recommendations#setting-requests-and-limits)Setting Requests and Limits
+
+❗ **If you set a resource limit, you should also set a resource request.** Otherwise the Request will match the limit. For example, a Deployment with _no_ defined CPU request and a defined CPU limit of 1 core will _result in a pod with a request of 1 CPU and a limit of 1 CPU_.
+
+**General Guidelines**  
+☑ Set requests and limits.  
+☑ Set requests to the _minimum_ of what your application needs.  
+☑ Set limits to a reasonable burstable number of what a single pod should support.  
+☑ Use horizontal pod autoscalers where possible, rather than large CPU and memory limits.
+
+**Being A Good Resource Citizen**
+
+⭐ ⭐ ⭐
+
+Having a **3:1 ratio** of CPU Limit:CPU Request is a good starting place for new applications that haven't yet been tuned. Using a 3:1 ratio makes you a **good community member**!
+
+⭐ ⭐ ⭐ ⭐
+
+Having a **2:1 ratio** of CPU Limit:CPU Request is a great next step for teams whose projects are working and stable, and who are in a position to start tuning their application more effectively - especially those who are seeking to make better use of horizontal scaling. Using a 2:1 ratio makes you a **great community member**!
+
+⭐ ⭐ ⭐ ⭐ ⭐
+
+Having a **1.5:1 ratio** of CPU Limit:CPU Request is an amazing goal for teams who have already started tuning their applications and are looking to make the best possible use of the platform's capabilities. Using a 1.5:1 ratio makes you an **amazing community member**!
+
+### CPU and Memory Utilization
+
+Here is a 4 minute video that includes an example of resource tuning for a sample OpenShift application.
+
+[https://youtu.be/rkxVZgn9icU](https://youtu.be/rkxVZgn9icU)
+
+### Resources
+
+**Deploying pods without specifying  a limit or a request** 
+
+If you deploy pods without setting limits or requests, they will be deployed with the following defaults:
+
+* CPU Request: 100m
+* CPU Limit: 250m
+* Memory Request: 256Mi
+* Memory Limit: 1Gi
+
+This is the NOT same as specifying a resource request or limit of 0\.
+
+**Setting the request and limits to 0**  
+If you set requests and limits to 0, pods will run under the BestEffort QoS class, using whatever spare capacity is available on the node.
```

Comment:
```suggestion
If you set requests and limits to 0, pods will run under the [BestEffort QoS class](https://docs.okd.io/4.9/applications/quotas/quotas-setting-per-project.html), using whatever spare capacity is available on the node.
```

Reply:
I'm not sure we need a link here? But if we do, https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/ might be a better option? The quotas page is more talking about scoping quotas to terminating/non-terminating pods.

Even then, we should use the OCP docs instead of OKD. https://docs.openshift.com/container-platform/4.9/applications/quotas/quotas-setting-per-project.html

Reply:
I think it having the link to define BestEffort Qos class and provide the extra detail is helpful, I've included your suggested link Steven.